Slim/Networking/Async/HTTP.pm

```diff
@@ -438,9 +438,7 @@ sub _http_read {
 		# Body might also be empty but a keep-alive with no content-length in the response is an error
 		# if everything has already been read, _http_body_read will unsubscribe to event loop 
 		# we just subscrive above ... a bit unefficient 
-		_http_read_body( $self->socket, $self, $args ) if ( !($self->response->headers->header('Connection') =~ /close/i || 
-															  $headers->header('Connection') !~ /keep-alive/i ) && 
-															$self->socket->_rbuf_length == ($headers->content_length || 0));
+		_http_read_body( $self->socket, $self, $args ) if ( $self->response->headers->header('Connection') =~ /keep-alive/i && $self->socket->_rbuf_length == ($headers->content_length || 0));
```

Comment:
I think you should test when there is not 'Connection' header at all because it's an optional header and by default HTTP 1.1 connections are persistent.